TLS config tests are flaky

**Describe the bug**
```
$ go clean -testcache
$ go test github.com/jaegertracing/jaeger/pkg/config/tlscfg
...
--- FAIL: TestReload_ca_certs (20.18s)
    cert_watcher_test.go:160:
        	Error Trace:	cert_watcher_test.go:160
        	Error:      	Should be true
        	Test:       	TestReload_ca_certs
--- FAIL: TestAddCertsToWatch_remove_ca (0.00s)
    cert_watcher_test.go:312:
        	Error Trace:	cert_watcher_test.go:312
        	Error:      	Should be true
        	Test:       	TestAddCertsToWatch_remove_ca
FAIL
FAIL	github.com/jaegertracing/jaeger/pkg/config/tlscfg	20.472s
FAIL
$ go test github.com/jaegertracing/jaeger/pkg/config/tlscfg
ok  	github.com/jaegertracing/jaeger/pkg/config/tlscfg	0.575s
```

**To Reproduce**
Steps to reproduce the behavior:
1. `go clean -testcache`
2. `go test github.com/jaegertracing/jaeger/pkg/config/tlscfg`
3. Able to reproduce the failed tests quite reliably, often on the first pass.

**Expected behavior**
Tests to pass.

**Screenshots**

**Version (please complete the following information):**
 - OS: MacOS
 - Jaeger version: master
 - Deployment: N/A

**What troubleshooting steps did you try?**
I suspect it's due to the `certWatcher` failing to detect file changes on the filesystem.

I tried modifying tests to ensure synchronous writes to the cert files and that seems to have alleviated the problem. That is, I no longer could see the failed tests when repeatedly (about 10 times) running:
```
$ go clean -testcache
$ go test github.com/jaegertracing/jaeger/pkg/config/tlscfg
```

**Additional context**

--- a/pkg/config/tlscfg/cert_watcher_test.go
+++ b/pkg/config/tlscfg/cert_watcher_test.go
@@ -316,6 +316,8 @@ func waitUntil(f func() bool, iterations int, sleepInterval time.Duration) {
 	}
 }
 
+// syncWrite ensures data is written to the given filename and flushed to disk.
+// This ensures that any watchers looking for file system changes can be reliably alerted.
 func syncWrite(filename string, data []byte, perm os.FileMode) error {
 	f, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC|os.O_SYNC, perm)
 	if err != nil {

--- a/pkg/config/tlscfg/cert_watcher_test.go
+++ b/pkg/config/tlscfg/cert_watcher_test.go
@@ -84,11 +84,11 @@ func TestReload(t *testing.T) {
 	// update the content with client certs
 	certData, err = ioutil.ReadFile(clientCert)
 	require.NoError(t, err)
-	err = ioutil.WriteFile(certFile.Name(), certData, 0644)
+	err = syncWrite(certFile.Name(), certData, 0644)
 	require.NoError(t, err)
 	keyData, err = ioutil.ReadFile(clientKey)
 	require.NoError(t, err)
-	err = ioutil.WriteFile(keyFile.Name(), keyData, 0644)
+	err = syncWrite(keyFile.Name(), keyData, 0644)
 	require.NoError(t, err)
 
 	waitUntil(func() bool {
@@ -138,11 +138,11 @@ func TestReload_ca_certs(t *testing.T) {
 	// update the content with client certs
 	caData, err = ioutil.ReadFile(caCert)
 	require.NoError(t, err)
-	err = ioutil.WriteFile(caFile.Name(), caData, 0644)
+	err = syncWrite(caFile.Name(), caData, 0644)
 	require.NoError(t, err)
 	clientCaData, err = ioutil.ReadFile(caCert)
 	require.NoError(t, err)
-	err = ioutil.WriteFile(clientCaFile.Name(), clientCaData, 0644)
+	err = syncWrite(clientCaFile.Name(), clientCaData, 0644)
 	require.NoError(t, err)
 
 	waitUntil(func() bool {
@@ -200,11 +200,11 @@ func TestReload_err_cert_update(t *testing.T) {
 	// update the content with client certs
 	certData, err = ioutil.ReadFile(badCaCert)
 	require.NoError(t, err)
-	err = ioutil.WriteFile(certFile.Name(), certData, 0644)
+	err = syncWrite(certFile.Name(), certData, 0644)
 	require.NoError(t, err)
 	keyData, err = ioutil.ReadFile(clientKey)
 	require.NoError(t, err)
-	err = ioutil.WriteFile(keyFile.Name(), keyData, 0644)
+	err = syncWrite(keyFile.Name(), keyData, 0644)
 	require.NoError(t, err)
 
 	waitUntil(func() bool {
@@ -314,4 +314,15 @@ func waitUntil(f func() bool, iterations int, sleepInterval time.Duration) {
 		}
 		time.Sleep(sleepInterval)
 	}
+}
+
+func syncWrite(filename string, data []byte, perm os.FileMode) error {
+	f, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC|os.O_SYNC, perm)
+	if err != nil {
+		return err
+	}
+	if _, err = f.Write(data); err != nil {
+		return err
+	}
+	return f.Sync()
 }

